src/dataExplorer/components/ResultsPane.tsx

```diff
@@ -119,9 +119,33 @@ const ResultsPane: FC = () => {
   }
 
   const downloadByServiceWorker = () => {
-    // TODO (chunchun): https://github.com/influxdata/ui/issues/6704
     if (language === LanguageType.INFLUXQL) {
-      console.error('csv download for InfluxQL is not implemented')
+      try {
+        event('runQuery.downloadCSV.influxql', {context: 'query experience'})
+
+        const params = new URLSearchParams({
+          db: selection.dbrp?.database,
+          q: text,
+        })
+        if (selection.dbrp?.retention_policy) {
+          params.set('rp', selection.dbrp?.retention_policy)
+        }
+        const url = `${API_BASE_PATH}query?${params}`
```

Comment:
this is a duplicate of the code on 155-160, perhaps you could think about making a reusable function